File: C++/0082. 猴子爬山.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.
#include <vector>
int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 
    int n;
    while(cin>>n) {
        vector<int> arr(n + 1, 1);
        for(int i=3;i<=n;++i) {
            arr[i] = arr[i - 1] + arr[i - 3];
        }
        cout<<arr[n]<<endl;
    }
    return 0;
}

File: Python/0082. 猴子爬山.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
            n = int(input())
            arr = [1] * (n + 1)
            
            for i in range(3, n+1):
                arr[i] = arr[i - 1] + arr[i - 3]
            print(arr[n])
            
        except EOFError:
            break

    
if __name__ == "__main__":
    func()
